### PR TITLE
Refactoring / searx / stackoverflow user page

### PR DESCRIPTION
**Pull request by artegoser**
- deleted public config, replaced with package.json. Updated version to 1.6.0 for this pull request.
- added type system for engine routes
- Search route replaced with universal route /redirect
- Added stackoverflow users parser
- Added NoHandlerFoundError for routes

### CODE REVIEW

File: src/handlers/engines/stackoverflow/users.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+      return `<strong>${type} (${votes}) </strong><a href="${url}">${title}</a>`;
+    })
+    .join('<br/>');
+
+  return {
+    content: `${userInfo}<hr><h3>Top Posts</h3>${topPosts}`,
+    textContent: `${ro.q.id}/${ro.q.slug}\n`,
+    title: document.querySelector('title')?.textContent || '',
+    lang: 'en',
```
Comment (DarkCat09) [inline src/handlers/engines/stackoverflow/users.ts:33]: Not always English, e.g. `ru.stackoverflow.com`. Maybe `document.documentElement.lang` as above?

File: tsconfig.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
     // "sourceMap": true,                                /* Create source map files for emitted JavaScript files. */
     // "inlineSourceMap": true,                          /* Include sourcemap files inside the emitted JavaScript. */
     // "outFile": "./",                                  /* Specify a file that bundles all outputs into one JavaScript file. If 'declaration' is true, also designates a file that bundles all .d.ts output. */
-    "outDir": "./dist/src/" /* Specify an output folder for all emitted files. */,
+    "outDir": "./dist/" /* Specify an output folder for all emitted files. */,
```
Comment (DarkCat09) [inline tsconfig.json:58]: Please, check if it does not break any build scripts including CI

**Review by DarkCat09 — COMMENTED** [review]
OK. Also, good solution with packageJSON

**Comment by DarkCat09** [comment]
`/redirect` instead of `/search` gives more flexibility, also good solution, thanks.

File: tsconfig.json (hunk 2 above)
Comment (artegoser) [reply tsconfig.json:58]: It used to build in one folder, due to the fact that it is now importing from package.json, it now creates the src folder automatically. In the build everything is the same as it was, but now package.json is added.

File: src/handlers/engines/searx.ts [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+async function search(
+  input: HandlerInput,
+  ro: Route<{ search: string; pageno?: string }>
+) {
   const document = input.parseDom().window.document;
-  const search = req.search;
-  const url = new URL(input.getUrl());
-  const page = parseInt(url.searchParams.get('pageno') || '1');
+  const search = ro.q.search;
+  const page = parseInt(ro.q.pageno || '1');
```
[the remark this answers is not in the document]
Comment (artegoser) [reply src/handlers/engines/searx.ts:13]: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

File: src/handlers/engines/stackoverflow/main.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Engine } from '../../engine';
+import questions from './questions';
+import users from './users';
+const SOE = new Engine('StackOverflow', [
```
[the remark this answers is not in the document]
Comment (artegoser) [reply src/handlers/engines/stackoverflow/main.ts]: fixed

File: src/handlers/engines/stackoverflow/users.ts (hunk 1 above)
Comment (artegoser) [reply src/handlers/engines/stackoverflow/users.ts]: fixed

**Review by DarkCat09 — APPROVED** [review]

**Pull request merged**